jenkins_pipelines/scripts/maintenance_json_generator.py

```diff
@@ -148,174 +152,160 @@
+                            for i in range(1, 100):
+                                new_id = f"{mi_id}-{i}" 
+                                if new_id not in custom_repositories[node]:
+                                    final_id = new_id
+                                    break
+                        # for each mi_id we have multiple repos sometimes for each node
+                        custom_repositories[node][final_id] = repo
+                    else:
+                        custom_repositories[node] = {mi_id: repo}
 
     # Format into json and print
     # Check that it's not empty and save to file
-    if finaldict:
-        with open('custom_repositories.json', 'w', encoding='utf-8') as f:
-            json.dump(finaldict, f, indent=2)
-    else:
-        print("Dictionary is empty, something went wrong")
+    if not custom_repositories:
+        print("Empty custom_repositories dictionary, something went wrong")
         sys.exit(1)
 
+    with open('custom_repositories.json', 'w', encoding='utf-8') as f:
+        json.dump(custom_repositories, f, indent=2)
+    
+def is_under_embargo(mi_id: str) -> bool:
+    # MI under embargo should have the attribute OBS:EmbargoDate set to the embargo end
+    cmd = f"meta attribute SUSE:Maintenance:{mi_id}"
+    output = osc_ibs_request(cmd)
+    xml_attributes = ET.fromstring(output)
 
-def create_url(mi_id, suffix):
-    link = ["http://download.suse.de/ibs/SUSE:/Maintenance:/", str(mi_id)]
-
-    if link[:-1] == str(mi_id):
-        link.append(suffix)
-        url = ''.join(link)
-    else:
-        link = ''.join(link)
-        url = str(link) + suffix
-    re = requests.get(url)
-    if re.ok:
-        return url
-
+    embargo_attribute = xml_attributes.find("./attribute[@name='EmbargoDate'][value]")
+    if not embargo_attribute:
+        print(f"MI {mi_id} is not under embargo")
+        return False
+    
+    # TODO: verify what's the actual date format used in the EmbargoDate attribute
+    embargo_end_date_str = embargo_attribute.find("./value").text
+    embargo_end_date = datetime.strptime(embargo_end_date_str, "%Y-%m-%d").date()
+    if embargo_end_date >= CURRENT_DATE:
+        print(f"MI {mi_id} is under embargo until {embargo_end_date}. Today is {CURRENT_DATE}")
+        return True
+    
+    print(f"MI {mi_id} embargo was lifted after {embargo_end_date}. Today is {CURRENT_DATE}")
+    return False
 
 def main():
-    args = parse_args()
-    if args.mi_ids is not None:
+    args: argparse.Namespace = parse_args()
+
+    if args.mi_ids:
         # Ignore spaces before or after the comma
-        mi_ids = [id.strip() for id in args.mi_ids.split(",")]
+        mi_ids: set[str] = { id.strip() for id in args.mi_ids.split(",") }
     else:
-        mi_ids = read_requests()
-    find_valid_repos(mi_ids, args.version)
+        mi_ids: set[str] = read_requests()
 
+    if args.embargo_check:
+        mi_ids: set[str] = { id for id in mi_ids if not is_under_embargo(id) }
```

Comment:
It's not good to reuse the same variable name with a different content in the same function.

Reply:
I see the point, but I'm not sure in this specific case it would  be better to have something like

```python
if args.embargo_check:
        no_embargo_mi_ids: set[str] = { id for id in mi_ids if not is_under_embargo(id) }
        find_valid_repos(no_embargo_mi_ids, args.version)
else:
        find_valid_repos(mi_ids, args.version)
```

just to avoid reassigning the variable, given that it'll not be used for anything else after the function call